EIPS/eip-2917.md

```diff
@@ -0,0 +1,165 @@
+We note that the smart contract can be triggered for every computation of on the following events: 	
+- whenever the productivity of a user changes (increase/decrease), 
+- whenever a user withdraws.
+
+## Motivation
+
+One of the main drawbacks of many DeFi projects is the reward distribution mechanism within the smart contract. In fact, there are two main mechanisms are adopted so far.
+1.	Distribution of rewards is only given when all users exit the contract
+2.	The project collects on-chain data, conducts calculation off-chain, and sends the results
+to the chain before starting rewards distribution accordingly
+
+The first approach conducts all calculation in an on-chain fashion, the cycle of its rewards distribution is too long. Furthermore, users need to remove their collateral before getting the rewards, which can be harmful for their rewards. The second approach is a semi-decentralized model since the main algorithm involves an off-chain computation. Therefore, the fairness and transparency properties cannot be reflected and this can even create the investment barrier for users.
+
+Since there is more DeFi projects coming out everyday, users could not find a proper way  to get to know:
+1) amount of interests he/she would get
+2) how the interest calculated
+3) what is his/her contribution compare to the overall
+
+By standardizing ERC3000, it abstracts the interface for interests generation process. Making wallet applications easier to collect each DeFi's metrics, user friendlier.
+
+## Specification
+
+Every ERC-3000 compliant contract must implement the ERC3000 and ERC20 interfaces (if necessary):
+
+```solidity
+interface IERC3000 is IERC20 {
+
+    /// @dev This emit when interests amount per block is changed by the owner of the contract.
+    /// It emits with the old interests amount and the new interests amount.
+    event InterestRatePerBlockChanged (uint oldValue, uint newValue);
+
+    /// @dev This emit when a users' productivity has changed
+    /// It emits with the user's address and the the value after the change.
+    event ProductivityIncreased (address indexed user, uint value);
+
+    /// @dev This emit when a users' productivity has changed
+    /// It emits with the user's address and the the value after the change.
+    event ProductivityDecreased (address indexed user, uint value);
+
+    
+    /// @dev Return the current contract's interests rate per block.
+    /// @return The amount of interests currently producing per each block.
+    function interestsPerBlock() external view returns (uint);
+
+    /// @notice Change the current contract's interests rate.
+    /// @dev Note the best practice will be restrict the gross product provider's contract address to call this.
+    /// @return The true/fase to notice that the value has successfully changed or not, when it succeed, it will emite the InterestRatePerBlockChanged event.
```

Comment:
```suggestion
    /// @return The true/false to notice that the value has successfully changed or not, when it succeed, it will emite the InterestRatePerBlockChanged event.
```